YDotNet/Infrastructure/TypeCache.cs

```diff
@@ -9,7 +9,7 @@ public T GetOrAdd<T>(nint handle, Func<nint, T> factory)
     {
         if (handle == nint.Zero)
         {
-            ThrowHelper.ArgumentException("Cannot create object for null handle.", nameof(handle));
+            throw new ArgumentException("Cannot create object for null handle.", nameof(handle));
```

Comment:
There is good reason for the throw helper. It is about performance and consistency. I know it is a little bit odd with the syntax. But the main reason is that functions with throw calls cannot be inlined: https://reubenbond.github.io/posts/dotnet-perf-tuning